### DISCUSSION TITLE
Feat: Code Companion Chat With Buffers

### DISCUSSION THREAD

**Discussion by KTSCode** [olimorris/codecompanion.nvim#57] [Ideas]
**Context:**
I'm transitioning to Code Companion from [NeoAI](https://github.com/Bryley/neoai.nvim), which has a command `NeoAIContextOpen` to pass either the current buffer or visually selected text into the prompt as context. I find the `CodeCompanionWithBuffers` command very useful as it includes multiple open buffers as context. However, I prefer chat-based interactions where I can ask questions and understand the code better before making any changes. Sometimes, my goal is not to modify the code at all, but to gain a deeper understanding of it.

**Feature Request:**
1. **Chat Mode with Buffer Context:**
   - Ability to open a chat that includes all open buffers as context.
   - Option for the context to automatically update every time a new prompt is sent.
   - A keybinding in the chat buffer to manually refresh the buffer context.
   - If the open buffer's text must be included in the chat buffer, automatically collapse it in a fold.

**Custom Action Alternative:**
For the time being, I plan to try DIY-ing this with a custom action using the examples in `RECIPES.md`. I'll update this issue once I get that working. I decided to create this issue before doing that in case I'm missing something and a simple config change could enable this functionality.

**Comment by olimorris** [comment]
Thanks so much for your feedback and I hope you're enjoying the plugin.

Some great points. Tonight, I'm probably going to push a new feature (once I've updated the README) where you can do:

 `:<',>'CodeCompanion @buffers what does this code do?`

which will open a chat buffer, send the open buffers as context, send any visually selected text and submit the prompt. I also have plans to be able to add buffers to an existing chat by adding something like `@buffers` and parsing the buffer with Tree-sitter.

Oh and you'll also be able to add a mapping to your own custom prompts and call them with `:CodeCompanion @my_action can you fix this bug?`

**Reply by KTSCode** [reply]
That's awesome thank you @olimorris  I would love to help test that out!

**Reply by olimorris** [reply]
I've just pushed an update and tweaked the README. The relevant section is Getting Started. Let me know what you think.
